packages/trace-viewer/src/sw/traceModelBackends.ts

```diff
@@ -129,14 +128,22 @@ export class FetchTraceModelBackend implements TraceModelBackend {
     const fileName = entries.get(entryName);
     if (!fileName)
       return;
-    return fetch('/trace/file?path=' + encodeURIComponent(fileName));
+
+    return fetch(formatTraceFileUrl(fileName));
   }
 }
 
-function formatUrl(trace: string) {
-  let url = trace.startsWith('http') || trace.startsWith('blob') ? trace : `file?path=${encodeURIComponent(trace)}`;
-  // Dropbox does not support cors.
-  if (url.startsWith('https://www.dropbox.com/'))
-    url = 'https://dl.dropboxusercontent.com/' + url.substring('https://www.dropbox.com/'.length);
-  return url;
+const baseURL = new URL(self.location.href);
+baseURL.port = baseURL.searchParams.get('testServerPort') ?? baseURL.port;
```

Comment:
I find this too intrusive. Most likely, there are more places than need to move `testServerPort` into some `url.port`. Can we avoid a separate `testServerPort`? For example, forward requests from our server to the vite server or something like that?